cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionController.cpp

```diff
@@ -0,0 +1,272 @@
+  m_conversionResults(new GenericListModel(&CoordinateConversionResult::staticMetaObject, this)),
+  m_geoView(nullptr),
+  m_inPickingMode(false)
+{
+  {
+    // Some default coordinate conversion formats to set us up with.
+    m_coordinateFormats->setDisplayPropertyName("name");
+    m_coordinateFormats->append(QList<QObject*>()
+      << createDecimalDegrees()
+      << createDegreesDecimalMinutes()
+      << createDegreesMinutesSeconds()
+      << createMgrs()
+      << createUsng()
+      << createUtm()
+      << createGars()
+    );
+  }
+
+  {
+    m_conversionResults->setDisplayPropertyName("name");
+    connect(m_conversionResults, &QAbstractItemModel::rowsInserted, this,
+      [this](const QModelIndex& /*parent*/, int first, int last)
+      {
+        // Every result owned by the controller needs to listen to the
+        // currentPointChanged signal to update its notation.
+        for (int i = first; i <= last; ++i)
+        {
+          auto index = m_conversionResults->index(i);
+          auto result = m_conversionResults->element<CoordinateConversionResult>(index);
+          if (result)
+            connect(this, QOverload<const Point&>::of(&CoordinateConversionController::currentPointChanged),
+                    result, QOverload<const Point&>::of(&CoordinateConversionResult::updateCoordinatePoint));
+          else
+            qWarning("Empty CoordinateConversionResult in the CoordinateConversionController results model at index %d", i);
+        }
+      }
+     );
+  }
+}
+
+CoordinateConversionController::~CoordinateConversionController()
+{
+}
+
+QObject* CoordinateConversionController::geoView() const
+{
+  return m_geoView;
+}
+
+void CoordinateConversionController::setGeoView(QObject* geoView)
+{
+  if (geoView == m_geoView)
+    return;
+
+  if (m_geoView)
+    disconnect(this, nullptr, m_geoView, nullptr);
+
+  m_geoView = geoView;
+
+  if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+  {
+    connect(sceneView, &SceneViewToolkit::mouseClicked, this,
+            [sceneView, this](QMouseEvent& event)
+    {
+      if (m_inPickingMode)
+      {
+        setCurrentPoint(sceneView->screenToBaseSurface(event.x(), event.y()));
+        event.accept();
+      }
+    });
+  }
+  else if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+  {
+    connect(mapView, &MapViewToolkit::mouseClicked, this,
+            [mapView, this](QMouseEvent& event)
+    {
+      if (m_inPickingMode)
+      {
+        setCurrentPoint(mapView->screenToLocation(event.x(), event.y()));
+        event.accept();
+      }
+    });
+  }
+
+  emit geoViewChanged();
+}
+
+void CoordinateConversionController::setCurrentPoint(const Point& p)
+{
+  if (p == m_currentPoint)
+    return;
+
+  m_currentPoint = p;
+  forceUpdateCoordinates();
+}
+
+void CoordinateConversionController::setCurrentPoint(
+  const QString& p,
+  CoordinateConversionOption* option)
+{
+  if (auto geoView = qobject_cast<GeoView*>(m_geoView))
```

Comment:
when would this be false?

Reply:
`m_geoView` is technically an arbitrary QObject.